crates/libtiny_tui/src/config.rs

```diff
@@ -28,6 +34,117 @@ pub(crate) struct Config {
     pub(crate) key_map: Option<KeyMap>,
 }
 
+impl Config {
+    /// Gets tab configs for `server`
+    /// Prioritizing configs under the server or using defaults
+    pub(crate) fn server_tab_configs(&self, server: &str) -> TabConfigs {
+        let server_config = self.servers.iter().find_map(|s| {
+            if s.addr == server {
+                Some(&s.configs)
+            } else {
+                None
+            }
+        });
+        self.defaults.tab_configs.merge(server_config)
+    }
+
+    /// Gets tab configs for `chan` in `server`
+    /// Prioritizing configs under the chan, then the server, then the defaults
+    pub(crate) fn chan_tab_configs(&self, server: &str, chan: &ChanNameRef) -> TabConfigs {
+        let tab_config = self
+            .servers
+            .iter()
+            .find(|s| s.addr == server)
+            .and_then(|s| {
+                s.join
+                    .iter()
+                    .find_map(|c| {
+                        if c.name() == chan.display() {
+                            Some(c.tab_configs())
+                        } else {
+                            None
+                        }
+                    })
+                    .flatten()
+            });
+        self.server_tab_configs(server).merge(tab_config)
+    }
+
+    pub(crate) fn user_tab_configs(&self) -> TabConfigs {
+        TabConfigs {
+            ignore: Some(false),
+            notifier: Some(Notifier::Messages),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub(crate) struct Server {
+    pub(crate) addr: String,
+    pub(crate) join: Vec<Chan>,
+    #[serde(flatten)]
+    pub(crate) configs: TabConfigs,
+}
+
+#[derive(Debug, Default, Deserialize, PartialEq)]
+pub(crate) struct Defaults {
+    #[serde(flatten)]
+    pub(crate) tab_configs: TabConfigs,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+#[serde(untagged)]
+pub(crate) enum Chan {
+    /// Channel specified by name only
+    Name(String),
+    /// Channel specified by name and extra configurations
+    WithConfigs {
+        name: String,
+        #[serde(flatten)]
+        configs: TabConfigs,
+    },
+}
+
+#[derive(Debug, Default, Copy, Clone, Deserialize, PartialEq)]
+pub(crate) struct TabConfigs {
```

Comment:
Maybe make this singular, `TabConfig`? Maybe that's just me but I read this as "configurations for multiple tabs", but it's for one tab.